pkg/virt-controller/watch/migration.go

```diff
@@ -410,6 +414,19 @@ func (c *MigrationController) canMigrateVMI(migration *virtv1.VirtualMachineInst
 
 }
 
+// flushPendingQueue checks if a migration just finished.
+// If it did, it will enqueue all items from the pendingQueue to the main queue and empty the pendingQueue.
+func (c *MigrationController) flushPendingQueue(oldPhase, newPhase virtv1.VirtualMachineInstanceMigrationPhase) {
```

Comment:
Can we add unit tests for this? @fossedihelm do we have fake client ready for migration controller?

Reply:
Yes, but PR requires review https://github.com/kubevirt/kubevirt/pull/11844